docs/pages/machine-id/architecture.mdx

```diff
@@ -0,0 +1,148 @@
+the user.
+
+These credentials can then be used to complete any action that is allowed by the
+role's configured permissions. 
+
+In the case of Machine ID, the bot user is assigned a bot role, which includes 
+permissions to impersonate the roles that the user has configured.
+
+## Onboarding
+
+Onboarding is the process by which `tbot` initially authenticates with the 
+Teleport auth service in order to receive credentials.
+
+Machine ID leverages the existing Token resource within Teleport, with the
+token containing an additional `botName` field that identifies the bot user
+associated with the token. 
+
+Machine ID currently supports two methods of joining that have some key
+differences.
+
+Static Token:
+
+- The name of the token is the only value needed to join. This value is a
+  secret and must be handled securely.
+- Once used, the token resource self-destructs. This means it can only be used 
+  to join a single bot to a Teleport cluster.
+- The certificates exchanged for the token are 
+  [renewable](#renewable-certificates).
+
+
+Dynamic join tokens (e.g AWS IAM):
+
+- These tokens rely on some kind of external system that allows the bot to
+  prove it is allowed to join the cluster. The name of the token identifies the
+  Token resource in Teleport that contains the configuration.
+- The token can be used to join as many bots as you want.
+- The certificates exchanged for the token are not renewable. When the bot wants
+  to renew its certificates, it simply repeats the original join steps.
+
+Where possible, you should prefer to use a dynamic join token over a static
+token as this eliminates the need to handle a secret.
+
+## Renewable certificates
+
+Human Teleport users are not able to renew their credentials without
+re-authenticating. As Machine ID must work without manual human intervention,
+bots that have joined using static tokens are granted credentials that are
+marked as renewable. This means that the bot can use these credentials to
+fetch a new set of credentials with an expiry date further in the future.
+
+In order to mitigate the risk of bot user credentials being stolen, and then
+continually renewed by a malicious actor, renewable bot user certificates
+include a **generation counter**.
+
+The generation counter is stored against the user in the database and within the
+certificate. This counter is incremented each time the user renews their 
+certificate. When a bot attempts to renew, the Auth Server ensures that the 
+value within the certificate and in the database match. If they do not match,
+then the bot user is automatically locked. This means that if certificates are
+stolen, and attempted to be renewed whilst the bot is still running, the next
+renewal will render them useless.
+
+## File permissions
+
+There are two types of folder in use by `tbot`:
+
+- The bot's own files: these store sensitive credentials belonging to the bot
+  user itself. This is by default stored at `/var/lib/teleport/bot/`.
```

Comment:
I think "bot user" could be a little misleading. Maybe something like this?

```suggestion
- The bot's own files: these store sensitive credentials belonging to the
  `tbot` process itself. This is by default stored at `/var/lib/teleport/bot/`.
```

It may also be worth mentioning what these credentials are (renewable, etc) to hopefully explain why they're sensitive?